src/Interface/hiopInterfacePrimalDecomp.cpp

```diff
@@ -0,0 +1,216 @@
+#include "hiopInterfacePrimalDecomp.hpp"
+
+
+using namespace hiop;
+hiopInterfacePriDecProblem::RecourseApproxEvaluator::
+RecourseApproxEvaluator(int nc) : RecourseApproxEvaluator(nc, nc)  //nc_ <= nx, nd=S
+{
+}
+
+hiopInterfacePriDecProblem::RecourseApproxEvaluator::
+RecourseApproxEvaluator(int nc, int S) 
+  : nc_(nc),S_(S),rval_(0.)//nc = nx, nd=S
+{
+  assert(S>=nc);
+  xc_idx_.resize(nc_);
+  for(int i=0;i<nc_;i++) xc_idx_[i] = i;
+  rgrad_ = new double[nc];
+  rhess_ = new double[nc];
+  x0_ = new double[nc];
+}
+
+hiopInterfacePriDecProblem::RecourseApproxEvaluator::
+RecourseApproxEvaluator(const int nc, 
+                        const int S, 
+                        const std::vector<int>& list)
+  : nc_(nc),S_(S),rval_(0.),rgrad_(NULL), rhess_(NULL),x0_(NULL)//nc = nx, nd=S
+{
+  rgrad_ = new double[nc];
+  rhess_ = new double[nc];
+  x0_ = new double[nc];
+  assert(list.size()==nc_);
+  xc_idx_ = list;
+}
+
+hiopInterfacePriDecProblem::RecourseApproxEvaluator::
+RecourseApproxEvaluator(const int nc, 
+                        const int S, 
+                        const double& rval, 
+                        const double* rgrad, 
+                        const double* rhess, 
+                        const double* x0)
+  : nc_(nc), S_(S)
+{
+  //assert(S>=nc);
+  rval_ = rval;
+  rgrad_ = new double[nc];
+  rhess_ = new double[nc];
+  x0_ = new double[nc];
+    
+  xc_idx_.resize(nc_);
+  for(int i=0;i<nc_;i++) xc_idx_[i] = i;
+
+  memcpy(rgrad_,rgrad, nc*sizeof(double));
+  memcpy(rhess_,rhess , nc*sizeof(double));
+  memcpy(x0_,x0, nc*sizeof(double));
```

Comment:
This is not portable solution.

Reply:
This part has been done updated with copyFrom function and rgrad, rhess, x0 are now hiopVectors.